### ISSUE TITLE
Remote-Container extension does not use Docker extension setting

### ISSUE THREAD

**Issue by shu-mutou** [microsoft/vscode-remote-release#3061]
- VSCode Version: 1.45.1
- Local OS Version: windows 10
- Remote OS Version: ubuntu 18.04
- Remote Extension/Connection Type: Docker
- Remote-Container extension Version: 0.117.1

To configure proxy setting, I added docker command templates for `docker build` into `settings.json` as following:
```
  "docker.commands.build": "docker build --build-arg HTTPS_PROXY=<my proxy url> --build-arg HTTP_PROXY=<my proxy url>  --rm -f \"${dockerfile}\" -t ${tag} \"${context}\""
```
according to follow:
https://code.visualstudio.com/docs/containers/reference#_docker-build

I confirmed that Docker extension itself uses above settings, but Remote-Container extension does not use it and failed "docker build" when I try `Remote-Containers: Try a Sample...`.

**Comment by shu-mutou** [comment]
Is there any way to pass `--build-arg` to `docker` from `settings.json`? I don't want to include proxy settings into source repository.

**Comment by Chuxel** [comment]
A way to do this is to use a local environment variable that is either passed in as a build arg or a container environment variable in devcontainer.json:

```
"build": {
    "dockerfile": "Dockerfile",
    "args": {
        "HTTPS_PROXY": "${localEnv:HTTPS_PROXY}"
    }
}
```

... or ...

```
"containerEnv": {
        "HTTPS_PROXY": "${localEnv:HTTPS_PROXY}"
}
```

**Comment by shu-mutou** [comment]
Got it thank you! However, unfortunately, we need more works to install CA for proxy into container. Could you give me any advices for CA install?

**Comment by PavelSosin-320** [comment]
If dev container is deployed as a service the solution is a part of the docker framework.  Put any json into docker config, install jq into base dev container, using `apk add jq `(Alpine !), create or update running service with config and parse config into the environment variables using jq from the vscode user's profile. This is the standard way to config services.

**Comment by chrmarti** [comment]
@shu-mutou The CA question seems like a general Docker / Linux question and is not specific to DevContainers. Best to ask on StackOverflow. Thanks.

**Comment by shu-mutou** [comment]
Also, I tried another way.
* Created git repository, added `.devcontainer` directory into it and setup the directory.
  + `devcontainer.json` has `build.dockerfile`,  `build.args.HTTPS_PROXY`
* Added Dockerfile
  + `FROM ubuntu`
  +  locally, install proxy settings and packages for installing CA provided by proxy.
* Ran `Remote-Containers: Open Repository in Container...` from vscode.

Before creating container from my `.devcontainer` settings, vscode remote container seems to create intermediate container with alpine image and tried to install nodejs, git, openssh client and docker cli into it. But it does not seems have proxy settings. So the intermediate container could not be built and failed to construct devcontainer.

**Comment by shu-mutou** [comment]
Do I need to customize `scripts/volumeBootstrap.Dockerfile` ?

**Comment by chrmarti** [comment]
@shu-mutou Could you append the log? (Dismiss the error dialog and copy from the Dev Container terminal.)

**Comment by shu-mutou** [comment]
@chrmarti Terminal logs are followings. thanks!

```
[0 ms] Start: Resolving remote
[9 ms] Start: Check Docker is running
[10 ms] Start: Run: docker info
[1100 ms] Stop (1090 ms): Run: docker info
[1101 ms] Stop (1092 ms): Check Docker is running
[1104 ms] Start: Run: docker volume ls -q
[2599 ms] Stop (1495 ms): Run: docker volume ls -q
[2601 ms] Start: Run: docker ps -q -a --filter label=vsch.local.repository=ssh://mutou@10.xxx.xx.xxx:/opt/repo/test --filter label=vsch.local.repository.volume=test-repo --filter label=vsch.local.repository.folder=nesbs3-test-repo --filter label=vsch.quality=stable
[3477 ms] Stop (876 ms): Run: docker ps -q -a --filter label=vsch.local.repository=ssh://mutou@10.xxx.xx.xxx:/opt/repo/test --filter label=vsch.local.repository.volume=test-repo --filter label=vsch.local.repository.folder=nesbs3-test-repo --filter label=vsch.quality=stable
[3479 ms] Start: Run: docker build -f d:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.117.1\scripts\volumeBootstrap.Dockerfile -t vsc-volume-bootstrap d:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.117.1\scripts

Sending build context to Docker daemon   7.68kB

fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz
WARNING: Ignoring http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz: network error (che
ck Internet connection and firewall)
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/x86_64/APKINDEX.tar.gz
```

**Comment by chrmarti** [comment]
@shu-mutou Have you tried configuring Docker with the proxy: https://docs.docker.com/network/proxy/#configure-the-docker-client?

**Comment by chrmarti** [comment]
That might not support adding a certificate. https://stackoverflow.com/a/28393663/7857763 shows how that can be added to the host Linux (the Docker Desktop VM seems to use `update-ca-certificates`.) To get access to the host Linux you can use `https://github.com/justincormack/nsenter1`.

Haven't tried that myself, let us know how it goes.

**Comment by shu-mutou** [comment]
> @shu-mutou Have you tried configuring Docker with the proxy: https://docs.docker.com/network/proxy/#configure-the-docker-client?

This works for `fetch http://dl-cdn.alpinelinux.org/....` and seems to success building image for `vsc-volume-bootstrap`. Because these `fetch` access via `http` and they do not use CA certs, I think.

Also, I found that I could not build images for any samples, e.g. `vscode-remote-try-php`. Because it uses `apt-get` and it uses `/etc/apt/apt.conf` for proxy setting, I think.

However, I found what I need to use vscode remote-container in our situation.

Thank you so much for helping!! ❤️
